scri/bms_transformations.py

```diff
@@ -0,0 +1,829 @@
+            
+    n_theta = 2 * ell_max + 1; n_phi = n_theta;
+            
+    f_LM = np.matmul(np.linalg.inv(np.array(from_SH_expansion)),np.array(from_zprime_to_z))
+    f_LM_padded = np.pad(f_LM, (0,(ell_max + 1)**2 - f_LM.size))
+    f_LM_S2 = spinsfast.salm2map(f_LM_padded, 0, ell_max, n_theta, n_phi)
+    
+    K_S2 = compute_conformal_factor(Lorentz, n_theta, n_phi)
+
+    result_S2 = K_S2**(Y_LM[0]) * f_LM_S2
+    
+    return spinsfast.map2salm(result_S2, 0, ell_max)
+
+def transform_function(f_LM, Lorentz, ell_max=None, tol=1e-15):
+    """Apply a Lorentz transformation to a spin-weight 0 funtion, where the function
+    is provided via it's spherical harmonic modes, the Ylms.
+
+    Parameters
+    ----------
+    f_LM: ndarray, dtype=complex
+        Ylms to be transformed.
+    Lorentz: Lorentz_transformation
+        Lorentz transformation to be used to transform the function.
+    ell_max: int
+        Maximum ell to use when expressing functions via coordinates on the two-sphere.
+        Defaults to the ell_max of the Lorentz transformation.
+    tol: float
+        Tolerance to use when rewriting the supertranslation's transformed Ylms
+        in terms of the untransformed Ylms. Defaults to 1e-15.
+    """
+    if ell_max is None:
+        ell_max = Lorentz.ell_max
+
+    f_LM_prime = np.zeros((ell_max + 1)**2, dtype=complex)
+    for L in range(0, int(np.sqrt(f_LM.size) - 1) + 1):
+        for M in range(-L, L+1):
+            if abs(f_LM[sf.LM_index(L, M, 0)]) < tol:
+                continue
+            f_LM_prime += f_LM[sf.LM_index(L, M, 0)] * transform_Y_LM((L, M), Lorentz, ell_max)
+
+    return f_LM_prime
+
+def transform_supertranslation(S, Lorentz, ell_max=None, tol=1e-15):
+    """Apply a Lorentz transformation to a supertranslation and multiply by
+    one over the conformal factor. This produces the supertranslation the appears
+    when commuting a supertranslation through a Lorentz transformation.
+
+    The Lorentz transformation is the transformation appearing on
+    the RHS of the product, i.e., S' = L^{-1} S L.
+
+    Parameters
+    ----------
+    S: ndarray, dtype=complex
+        supertranslation to be transformed.
+    Lorentz: Lorentz_transformation
+        Lorentz transformation to be used to transform the supertranslation.
+    ell_max: int
+        Maximum ell to use when expressing functions via coordinates on the two-sphere.
+    tol: float
+        Tolerance to use when rewriting the supertranslation's transformed Ylms
+        in terms of the untransformed Ylms. Defaults to 1e-15.
+    """
+    if ell_max is None:
+        ell_max = Lorentz.ell_max
+    n_theta = 2 * ell_max + 1; n_phi = n_theta;
+        
+    transformed_S = transform_function(S, Lorentz, ell_max, tol)
+    
+    transformed_S_S2 = spinsfast.salm2map(transformed_S, 0, ell_max, n_theta, n_phi)
+    K_S2 = compute_conformal_factor(Lorentz, n_theta, n_phi)
+
+    S_prime_S2 = K_S2**(-1)*transformed_S_S2
```

Comment:
Why isn't this just:
```suggestion
    S_prime_S2 = transformed_S_S2 / K_S2
```